compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -670,7 +670,7 @@ class TreeUnpickler(reader: TastyReader,
         readByte()
         val end = readEnd()
         val tp = readType()
-        val lazyAnnotTree = readLaterWithOwner(end, rdr => ctx => rdr.readTerm()(ctx))
+        val lazyAnnotTree = readLaterWithOwner(end, rdr => implicit ctx => rdr.readTerm())
```

Comment:
I propose we go to the new syntax `(using ctx) => rdr.readTerm()`.

Reply:
That doesn't work because the expected type is not a context function type but a regular function type, and I was hesitant to change this since it'll make it harder to port code between the Scala 2 and 3 readers.

Reply:
OK